pkg/tide/tide.go

```diff
@@ -280,6 +289,63 @@ var (
 		}, []string{
 			"controller",
 		}),
+		retests: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: "tide_retests_total",
+			Help: "Total number of test retriggers by org, repo, and branch. Incremented when Tide triggers tests for PRs that need retesting.",
+		}, []string{
+			"org",
+			"repo",
+			"branch",
+		}),
+		retestsByAction: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: "tide_retests_by_action_total",
+			Help: "Total number of retests by action type (TRIGGER for serial, TRIGGER_BATCH for batch). Helps identify whether batch or serial testing is causing more retests.",
+		}, []string{
+			"org",
+			"repo",
+			"branch",
+			"action",
+		}),
+		batchFailures: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: "tide_batch_failures_total",
+			Help: "Total number of times a batch test completes but PRs move back to missing state (indicating batch failure). This is expensive as all PRs in the batch need retesting.",
+		}, []string{
+			"org",
+			"repo",
+			"branch",
+		}),
+		poolMissingPRs: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "tide_pool_missing_prs",
+			Help: "Number of PRs with missing or failed tests in each pool. High values indicate testing bottlenecks.",
```

Comment:
What is "missing or failed tests" supposed to mean? A PR is in a merge pool only if it is passing tests.